sanic/server.py

```diff
@@ -154,6 +157,8 @@ def data_received(self, data):
             self.headers = []
             self.parser = HttpRequestParser(self)
 
+        self.state['requests_count'] = self.state['requests_count'] + 1
```

Comment:
I have seen some weird exceptions caused by `HttpRequestParser`, so to be safe I think it should be incremented before parsing the request.

https://docs.python.org/3/library/asyncio-protocol.html#asyncio.Protocol.data_received

> data_received() can be called an arbitrary number of times during a connection

So it's incorrect to incremented it every time in `data_received`